plugins/catalog-backend-module-gerrit/src/providers/GerritEntityProvider.ts

```diff
@@ -179,10 +184,46 @@ export class GerritEntityProvider implements EntityProvider {
     logger.info(`Found ${locations.length} locations.`);
   }
 
-  private createLocationSpec(project: string): LocationSpec {
+  private async createLocationSpec(project: string): Promise<LocationSpec> {
+    // If a branch has been configured, we can use it directly
+    if (this.config.branch) {
+      return {
+        type: 'url',
+        target: `${this.integration.config.gitilesBaseUrl}/${project}/+/refs/heads/${this.config.branch}/${this.config.catalogPath}`,
+        presence: 'optional',
+      };
+    }
+
+    // Else we call Gerrit API to know on which branch HEAD is pointing to
+    let response: Response;
+    const baseProjectApiUrl = getGerritProjectsApiUrl(this.integration.config);
+    const projectGetHeadUrl = `${baseProjectApiUrl}${encodeURIComponent(
```

Comment:
We do something similar in our gerrit setup.

You can use the "[branches](https://gittools.se.axis.com/gerrit/Documentation/rest-api-projects.html#get-branch)" api here to get actual json instead of the weird string repsonse, like: 

```
const projectGetHeadUrl = `${baseProjectApiUrl}${encodeURIComponent(
      project,
    )}/branches/HEAD`;
...
const gerritHeadResponse = (await parseGerritJsonResponse(
   response as any,
 )) as { ref: string; revision: string };
```
Returns actual json:
```
  HTTP/1.1 200 OK
  Content-Disposition: attachment
  Content-Type: application/json; charset=UTF-8

  )]}'
  {
    "ref": "refs/heads/master",
    "revision": "67ebf73496383c6777035e374d2d664009e2aa5c"
  }
```
vs

```
  HTTP/1.1 200 OK
  Content-Disposition: attachment
  Content-Type: application/json; charset=UTF-8

  )]}'
  "refs/heads/master"
```

Reply:
IMHO the [GET HEAD endpoint](https://gerrit-review.googlesource.com/Documentation/rest-api-projects.html#get-head) is better suited for this, since its description is exactly what we are trying to achieve here:
> Retrieves for a project the name of the branch to which HEAD points.

It returns a valid JSON value (after removing the `)]}'\n` prefix): it's not a JSON object but a JSON string.

Also, on our side we don't have the `refs/heads/` prefix in the `/branches/HEAD` endpoint response.

```
GET /branches/HEAD
)]}'
{"ref":"HEAD","revision":"delivery"}
```

What do you think?

Reply:
You are correct!  TIL about JSON strings.